Release notes May 13 2024 (#4318)
Azure/AKS: CHANGELOG.md

# Azure Kubernetes Service Changelog

## Release 2024-05-13

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/).

### Announcements
* Starting 1.30 Kubernetes version and 1.27 LTS versions, beta apis will be disabled by default, when you upgrade to [them](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli#before-you-begin). There will be an option provided to explicitly enable beta apis closer to the 1.30 release.
* In 2020 Docker enacted a Rate Limiting policy for all users. In-order to assist customers with the change, Microsoft worked directly with Docker to prevent users of Microsoft Azure from being impacted. However, beginning on June 30th, 2024, Azure customers will begin to be impacted by this limit. In-order for customers to mitigate the potential effects of this limit. We recommend customers begin to use the Artifact Cache feature within Azure Container Registry or sign up for a Docker Subscription.  More information is available [here](https://techcommunity.microsoft.com/t5/apps-on-azure-blog/best-practices-for-using-azure-container-registry-and-docker-hub/ba-p/4068979).

> * If you use any programming/scripting logic to list and select a minor version of Kubernetes before creating clusters with the `ListKubernetesVersions` API, note that starting from Kubernetes v1.27, the API returns `SupportPlan` as `[KubernetesOfficial, AKSLongTermSupport]`. Please ensure you update any logic to exclude `AKSLongTermSupport` versions to avoid any breaks and choose `KubernetesOfficial` support plan versions.  Otherwise, if LTS is indeed your path forward please first opt-into the Premium tier and the `AKSLongTermSupport` support plan versions from the `ListKubernetesVersions` API before creating clusters.  Refer [long term support](https://learn.microsoft.com//azure/aks/long-term-support) for more information.

Reviewer: >  If you use any programming/scripting logic to list and select a minor version of Kubernetes before creating clusters with the `ListKubernetesVersions` API, note that starting from Kubernetes v1.27, the API returns `SupportPlan` as `[KubernetesOfficial, AKSLongTermSupport]`. Please ensure you update any logic to exclude `AKSLongTermSupport` versions to avoid any breaks and choose `KubernetesOfficial` support plan versions.  Otherwise, if LTS is indeed your path forward please first opt-into the Premium tier and the `AKSLongTermSupport` support plan versions from the `ListKubernetesVersions` API before creating clusters.  Refer [long term support](https://learn.microsoft.com//azure/aks/long-term-support) for more information

Can we remove this one? We don't need it anymore.

Reviewer 2: yes pls.

  * Generally Available - [Security patch channel](https://learn.microsoft.com/en-us/azure/aks/auto-upgrade-node-os-image?tabs=azure-cli) for VHD updates.
  * Public Preview - Deployment Safeguards now supports [mutations in Enforcement mode](https://learn.microsoft.com/en-us/azure/aks/deployment-safeguards). 
  * Public Preview - [Native sidecars](https://learn.microsoft.com/en-us/azure/aks/istio-native-sidecar) under Istio-based service mesh addon in AKS.
  * Public Preview - [AKS Automatic](https://learn.microsoft.com/en-us/azure/aks/intro-aks-automatic).
  * Public Preview - [Node Initialization Taints](https://learn.microsoft.com/en-us/azure/aks/use-node-taints#use-node-initialization-taints-preview)
 
* Behavioral Changes:
  * Allow disabling NPM for existing clusters with "networkPolicy=none" for stable api version 2024-05-01. 
  * Minimum supported k8s version for advancedNetworking is 1.21.0 and minimum supported k8s version for cilium clusters with advancedNetworking is 1.29.0.
  * [Node upgrade (reimage)](https://learn.microsoft.com/en-us/azure/aks/auto-upgrade-node-os-image) will wait for disk detach to complete (to prevent very slow disk detach).

>   * Default network policy is "networkPolicy=none" when network policy is not set on new clusters starting from API version 2024-05-01.

Reviewer: Documentation link?

  * Block GetOsOptions operation in 2024-05-01/2024-05-02-preview API or later - [Github Issue 4261](https://github.com/Azure/AKS/issues/4261).
  * [Advanced Container Netwrking Services](https://learn.microsoft.com/en-us/azure/aks/advanced-container-networking-services-overview) can only be enabled on cilium clusters with k8s v1.29.0 onwards.
  * Customized apiserver subnet must be empty when migrating a cluster to enable [apiserver-vnet-integration](https://learn.microsoft.com/azure/aks/api-server-vnet-integration). If the subnet has resources in it, the migration won't be allowed.
   
* Bug Fixes:
  * Metrics server v0.6.3 will be used to prevent frequent OOMKills on Metric server.
  * Allowing zonal NodeClaims to facilitate NodeClaims and node creation on [Node Auto Provisioning](https://learn.microsoft.com/azure/aks/node-autoprovision) for workloads with zone affinity constraints.

>   * SSHAccess of nodepool will not be reset from Disabled to LocalUser by partial put. Put operation will be blocked with 400 error code.

Reviewer: Fixed a bug where the SSHAccess property of a node pool would be reset to LocalUser on a partial put.

# Azure Kubernetes Service Changelog

## Release 2024-05-13

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/).

### Announcements
* Starting 1.30 Kubernetes version and 1.27 LTS versions, beta apis will be disabled by default, when you upgrade to [them](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli#before-you-begin). There will be an option provided to explicitly enable beta apis closer to the 1.30 release.

> * On 15 March 2027, Windows Server 2022 will be retired when Kubernetes 1.34 reaches the end of platform support. You won't be able to create new Windows Server 2022 node pools on Kubernetes 1.35 and above. We encourage you to make the switch before 15 March 2027 to gain the richer benefits of [Windows Server 2025](https://techcommunity.microsoft.com/t5/windows-server-news-and-best/introducing-windows-server-2025/ba-p/4026374) or Windows Server [Annual Channel](https://techcommunity.microsoft.com/t5/windows-server-news-and-best/windows-server-annual-channel-for-containers/ba-p/3866248). These new Windows OS versions will be supported on AKS before Windows Server 2022 is retired. For more updates, see our [AKS public roadmap](https://github.com/Azure/AKS/projects/1).

Reviewer 3: ```suggestion
```
We can remove this, it's been on release notes for a while

* If you use any programming/scripting logic to list and select a minor version of Kubernetes before creating clusters with the `ListKubernetesVersions` API, note that starting from Kubernetes v1.27, the API returns `SupportPlan` as `[KubernetesOfficial, AKSLongTermSupport]`. Please ensure you update any logic to exclude `AKSLongTermSupport` versions to avoid any breaks and choose `KubernetesOfficial` support plan versions.  Otherwise, if LTS is indeed your path forward please first opt-into the Premium tier and the `AKSLongTermSupport` support plan versions from the `ListKubernetesVersions` API before creating clusters.  Refer [long term support](https://learn.microsoft.com//azure/aks/long-term-support) for more information.

> * [GetOSOptions](https://github.com/Azure/AKS/issues/4261) will no longer be included in new AKS API versions starting with 2024-05-01. This API was used to get OS options that support Federal Information Process Standard (FIPS) in the specified subscription. If you're calling this API via the CLI, it will no longer be available in newer az aks extension versions. You can use an older version of the az aks extension, however this is not recommended. For details on what AKS supported operating systems support Federal Information Process Standard (FIPS), see [aka.ms/aks/GetFIPSOSOptions](https://aka.ms/aks/GetFIPSOSOptions).

Reviewer 4: Can we specify the aks CLI module version from which it won't be included anymore. And we should add a link to updating Azure CLI or the aks-preview CLI extensions

Author: @allyford

Reviewer 3: Checking on cli module version

### Release Notes

> * Features:

Reviewer 4: In the past, we had 2 sections - Features and Preview Features. can we change to that format and move all GA features under Features and preview stuff under Preview Features section

> * If you use any programming/scripting logic to list and select a minor version of Kubernetes before creating clusters with the `ListKubernetesVersions` API, note that starting from Kubernetes v1.27, the API returns `SupportPlan` as `[KubernetesOfficial, AKSLongTermSupport]`. Please ensure you update any logic to exclude `AKSLongTermSupport` versions to avoid any breaks and choose `KubernetesOfficial` support plan versions.  Otherwise, if LTS is indeed your path forward please first opt-into the Premium tier and the `AKSLongTermSupport` support plan versions from the `ListKubernetesVersions` API before creating clusters.  Refer [long term support](https://learn.microsoft.com//azure/aks/long-term-support) for more information.

Reviewer 3: ```suggestion
* If you use any programming/scripting logic to list and select a minor version of Kubernetes before creating clusters with the `ListKubernetesVersions` API, note that starting from Kubernetes v1.27, the API returns `SupportPlan` as `[KubernetesOfficial, AKSLongTermSupport]`. Please ensure you update any logic to exclude `AKSLongTermSupport` versions to avoid any breaks and choose `KubernetesOfficial` support plan versions.  Otherwise, if LTS is indeed your path forward please first opt-into the Premium tier and the `AKSLongTermSupport` support plan versions from the `ListKubernetesVersions` API before creating clusters.  Refer [long term support](https://learn.microsoft.com//azure/aks/long-term-support) for more information.
* [GetOSOptions](https://github.com/Azure/AKS/issues/4261) will no longer be included in new AKS API versions starting with 2024-05-01. This API was used to get OS options that support Federal Information Process Standard (FIPS) in the specified subscription. If you're calling this API via the CLI, it will no longer be available in newer az aks extension versions. You can use an older version of the az aks extension, however this is not recommended. For details on what AKS supported operating systems support Federal Information Process Standard (FIPS), see [aka.ms/aks/GetFIPSOSOptions](https://aka.ms/aks/GetFIPSOSOptions).
```

>   * Generally Available - AKS supports [disabling Windows OutboundNAT](https://learn.microsoft.com/en-us/azure/aks/nat-gateway#disable-outboundnat-for-windows).

Reviewer 3: ```suggestion
  * Generally Available - AKS supports [disabling Windows OutboundNAT](https://learn.microsoft.com/azure/aks/nat-gateway#disable-outboundnat-for-windows).
```

>   * Generally Available - [Automated Deployments](https://learn.microsoft.com/en-us/azure/aks/automated-deployments).

Reviewer 3: ```suggestion
  * Generally Available - [Automated Deployments](https://learn.microsoft.com/azure/aks/automated-deployments).
```

>   * Generally Available - [Security patch channel](https://learn.microsoft.com/en-us/azure/aks/auto-upgrade-node-os-image?tabs=azure-cli) for VHD updates.

Reviewer 3: ```suggestion
  * Generally Available - [Security patch channel](https://learn.microsoft.com/azure/aks/auto-upgrade-node-os-image?tabs=azure-cli) for VHD updates.
```

>   * Public Preview - Deployment Safeguards now supports [mutations in Enforcement mode](https://learn.microsoft.com/en-us/azure/aks/deployment-safeguards). 

Reviewer 3: ```suggestion
  * Public Preview - Deployment Safeguards now supports [mutations in Enforcement mode](https://learn.microsoft.com/azure/aks/deployment-safeguards). 
```

>   * Public Preview - [Native sidecars](https://learn.microsoft.com/en-us/azure/aks/istio-native-sidecar) under Istio-based service mesh addon in AKS.

Reviewer 3: ```suggestion
  * Public Preview - [Native sidecars](https://learn.microsoft.com/azure/aks/istio-native-sidecar) under Istio-based service mesh addon in AKS.
```

>   * Public Preview - [AKS Automatic](https://learn.microsoft.com/en-us/azure/aks/intro-aks-automatic).

Reviewer 3: ```suggestion
  * Public Preview - [AKS Automatic](https://learn.microsoft.com/azure/aks/intro-aks-automatic).
```

>   * Public Preview - [Node Initialization Taints](https://learn.microsoft.com/en-us/azure/aks/use-node-taints#use-node-initialization-taints-preview)

Reviewer 3: ```suggestion
  * Public Preview - [Node Initialization Taints](https://learn.microsoft.com/azure/aks/use-node-taints#use-node-initialization-taints-preview)
```

 
* Behavioral Changes:

>   * Allow disabling NPM for existing clusters with "networkPolicy=none" for stable api version 2024-05-01. 

Reviewer 3: Is "disable NPM for existing clusters" added to our documentation? Can we add a link?

>   * Minimum supported k8s version for advancedNetworking is 1.21.0 and minimum supported k8s version for cilium clusters with advancedNetworking is 1.29.0.

Reviewer 3: In documentation?

Reviewer: Suggest changing k8s -> Kubernetes and linking to a documentation article.

>   * Disk Detach must complete before [node image auto upgrade](https://learn.microsoft.com/en-us/azure/aks/auto-upgrade-node-os-image) starts.

Reviewer 5: Node upgrade (reimage) will wait for disk detach (to prevent very slow disk detach)

>   * Disk Detach must complete before [node image auto upgrade](https://learn.microsoft.com/en-us/azure/aks/auto-upgrade-node-os-image) starts.

Reviewer 3: ```suggestion
  * Disk Detach must complete before [node image auto upgrade](https://learn.microsoft.com/azure/aks/auto-upgrade-node-os-image) starts.
```

>   * Default network policy is "networkPolicy=none" when network policy is not set on new clusters starting from API version 2024-05-01.

Reviewer 3: ```suggestion
  * Default network policy is "networkPolicy=none" when network policy is not set on new clusters starting from API version 2024-05-01. This means that Network Policy Manager (NPM) is disabled.
```
Is there an alternative that is enabled by default?

>   * Block GetOsOptions operation in 2024-05-01/2024-05-02-preview API or later - [Github Issue 4261](https://github.com/Azure/AKS/issues/4261).

Reviewer 3: ```suggestion
```
Move to announcements since it's a retirement.

>   * [Advanced Container Netwrking Services](https://learn.microsoft.com/en-us/azure/aks/advanced-container-networking-services-overview) can only be enabled on cilium clusters with k8s v1.29.0 onwards.

Reviewer 3: ```suggestion
  * [Advanced Container Netwrking Services](https://learn.microsoft.com/azure/aks/advanced-container-networking-services-overview) can only be enabled on cilium clusters with k8s v1.29.0 onwards.
```
I'm not seeing this limitation detailed in the documentation you linked. Can you reach out to the owner to properly document this limitation?

Reviewer: Advanced Container **Netwrking** Services -> Networking (spelling)

  * Customized apiserver subnet must be empty when migrating a cluster to enable [apiserver-vnet-integration](https://learn.microsoft.com/azure/aks/api-server-vnet-integration). If the subnet has resources in it, the migration won't be allowed.
   

> * Bug Fixes:

Reviewer 2: See if there is any GH link for the bugs wherever applicable in all the 3 points.

>   * Customized apiserver subnet must be empty when migrating a cluster to enable [apiserver-vnet-integration](https://learn.microsoft.com/en-us/azure/aks/api-server-vnet-integration). If the subnet has resources in it, the migration won't be allowed.

Reviewer 3: ```suggestion
  * Customized apiserver subnet must be empty when migrating a cluster to enable [apiserver-vnet-integration](https://learn.microsoft.com/azure/aks/api-server-vnet-integration). If the subnet has resources in it, the migration won't be allowed.
```

  * Metrics server v0.6.3 will be used to prevent frequent OOMKills on Metric server.

>   * Allowing zonal NodeClaims to facilitate NodeClaims and node creation on [Node Auto Provisioning](https://learn.microsoft.com/en-us/azure/aks/node-autoprovision) for workloads with zone affinity constraints.

Reviewer 3: ```suggestion
  * Allowing zonal NodeClaims to facilitate NodeClaims and node creation on [Node Auto Provisioning](https://learn.microsoft.com/azure/aks/node-autoprovision) for workloads with zone affinity constraints.
```

  * SSHAccess of nodepool will not be reset from Disabled to LocalUser by partial put. Put operation will be blocked with 400 error code.

>   * Fixed bug where the eTag property in 2024-02-02preview, 2024-03-02-preview, and 2024-04-02-preview APIs was returned with the wrong case (returned etag, should have been eTag).

Reviewer 3: ```suggestion
  * Fixed bug where the eTag property in 2024-02-02 preview, 2024-03-02 preview, and 2024-04-02 preview APIs was returned with the wrong case (returned etag, should have been eTag).
```

  * Patch version 1.28.9 fixes [Bug - OpenAPI handler fails on duplicated path](https://github.com/kubernetes/kubernetes/issues/122668). 

* Component Updates: 

>   * Istio-based service mesh add-on revision asm-1-19 has been upgraded to patch v1.19.10, asm-1-20 has been upgraded to patch v.1.20.6, and asm-1-21 has been upgraded to patch v1.21.2. Users can restart the workload pods to trigger re-injection of the newer patch version of istio-proxy. More information can be found [here](https://learn.microsoft.com/en-us/azure/aks/istio-upgrade).

Reviewer 3: ```suggestion
  * Istio-based service mesh add-on revision asm-1-19 has been upgraded to patch v1.19.10, asm-1-20 has been upgraded to patch v.1.20.6, and asm-1-21 has been upgraded to patch v1.21.2. Users can restart the workload pods to trigger re-injection of the newer patch version of istio-proxy. More information can be found [here](https://learn.microsoft.com/azure/aks/istio-upgrade).
```

>   * Linux and Windows  addon-token-adapter image is updated to mcr.microsoft.com/aks/msi/addon-token-adapter:master.240510.2

Reviewer 3: ```suggestion
  * Linux and Windows addon-token-adapter image is updated to mcr.microsoft.com/aks/msi/addon-token-adapter:master.240510.2
```
(nit)

>   * Preview API 2021-11-01 preview will be removed.

Author: this seems to have been regular cleanup so removing it

  * Managed Prometheus image version updated to [images:6.8.12-main-05-21-2024](https://github.com/Azure/prometheus-collector/blob/main/RELEASENOTES.md).

>   * Update Policy addon to v1.4.

Reviewer 3: Link?

>   * Update cloud node manager to v1.30.0 on AKS 1.30+, v1.29.4 on AKS 1.29+, v1.28.9 on AKS 1.28+, v1.27.17 on AKS 1.27+.

Reviewer 3: Link?

  * Update [aks app routing operator image](https://github.com/Azure/aks-app-routing-operator/blob/main/CHANGELOG.md#023---2024-04-22) to v0.2.3.

>   * Upgrade Azure File CSI driver to v1.28.10  on AKS 1.27, v1.29.5  on AKS 1.28, v1.30.2  on AKS 1.29.

Reviewer 3: Link?

  * AKS Ubuntu 22.04 image has been updated to [AKSUbuntu-2204-202405.20.0](vhd-notes/aks-ubuntu/AKSUbuntu-2204/202405.20.0.txt).
  * AKS Ubuntu 18.04 image has been updated to [AKSUbuntu-1804-202405.20.0](vhd-notes/aks-ubuntu/AKSUbuntu-1804/202405.20.0.txt).
  * Azure Linux image has been updated to [AzureLinux-202405.20.0](vhd-notes/AzureLinux/202405.20.0.txt).
  * AKS Windows Server 2019 image has been updated to [AKSWindows-2019-17763.5820.240516](vhd-notes/AKSWindows/2019/17763.5820.240516.txt).
  * AKS Windows Server 2022 image has been updated to [AKSWindows-2022-20348.2461.240516](vhd-notes/AKSWindows/2022/20348.2461.240516.txt).

> 

Reviewer 2: We can also add at the bottom - for up to date information on the latest images avaiable across platforms please go to https://releases.aks.azure.com/webpage/index.html

Reviewer 6: Does the release tracker get updated as soon as this release goes live with the latest versions? 

If not it would be nice for maybe the release tracker to show the upcoming releases or keep the new versions here as the information is helpful.

# Azure Kubernetes Service Changelog

## Release 2024-05-13

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/).

> ### Announcements

Reviewer 4: @palma21 I know it's not technically linked with a particular release, but should we add a line to the new AKS blog to increase awareness on a good source for more info to go along with release notes and docs?

Reviewer 4: Ah I missed a line on it. Never mind

  * Linux and Windows addon-token-adapter image for Azure monitoring metrics is updated to mcr.microsoft.com/aks/msi/addon-token-adapter:master.240510.2. The updated image patches [CVE-2023-4911](https://www.cve.org/CVERecord?id=CVE-2023-4911),[CVE-2024-2961](https://www.cve.org/CVERecord?id=CVE-2024-2961), [CVE-2024-33599](https://www.cve.org/CVERecord?id=CVE-2024-33599), [CVE-2024-33600](https://www.cve.org/CVERecord?id=CVE-2024-33600), [CVE-2024-33601](https://www.cve.org/CVERecord?id=CVE-2024-33601), [CVE-2024-33602](https://www.cve.org/CVERecord?id=CVE-2024-33602), [CVE-2023-3446](https://www.cve.org/CVERecord?id=CVE-2023-3446), [CVE-2023-3817](https://www.cve.org/CVERecord?id=CVE-2023-3817), [CVE-2023-3446](https://www.cve.org/CVERecord?id=CVE-2023-3446), [CVE-2023-3817](https://www.cve.org/CVERecord?id=CVE-2023-3817). 
  * Managed Prometheus image version updated to [images:6.8.12-main-05-21-2024](https://github.com/Azure/prometheus-collector/blob/main/RELEASENOTES.md).

>   * Update [Policy addon to v1.4](https://learn.microsoft.com/azure/governance/policy/concepts/policy-for-kubernetes#add-on-versions-available-per-each-aks-cluster-version) for all clusters on kubernetes version >= v1.25.

Reviewer 4: ```suggestion
  * Azure Policy addon has been updated to [v1.4.0](https://learn.microsoft.com/azure/governance/policy/concepts/policy-for-kubernetes#140) for all clusters on Kubernetes version >= v1.25.
```

>   * Update cloud node manager to v1.30.0 on AKS 1.30+, v1.29.4 on AKS 1.29+, v1.28.9 on AKS 1.28+, v1.27.17 on AKS 1.27+. Refer [AKS version matrix for cloud node manager](https://github.com/kubernetes-sigs/cloud-provider-azure#aks-version-matrix)

Reviewer 4: ```suggestion
  * Updated cloud node manager to [v1.30.0](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.30.0) on AKS 1.30+, [v1.29.4](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.29.4) on AKS 1.29+, [v1.28.9](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.28.9) on AKS 1.28+, [v1.27.17](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.27.17) on AKS 1.27+. Refer [AKS version matrix for cloud node manager](https://github.com/kubernetes-sigs/cloud-provider-azure#aks-version-matrix)
```

>   * Update [aks app routing operator image](https://github.com/Azure/aks-app-routing-operator/blob/main/CHANGELOG.md#023---2024-04-22) to v0.2.3.

Reviewer 4: ```suggestion
  * Updated AKS App Routing operator image to [v0.2.3](https://github.com/Azure/aks-app-routing-operator/blob/main/CHANGELOG.md#023---2024-04-22).
```

>   * Upgrade Azure File CSI driver to v1.28.10  on AKS 1.27, v1.29.5  on AKS 1.28, v1.30.2  on AKS 1.29. Refer -[Azure File CSI driver](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases)

Reviewer 4: ```suggestion
  * Updated Azure File CSI driver to [v1.28.10](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.28.10)  on AKS 1.27, [v1.29.5](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.29.5)  on AKS 1.28, [v1.30.2](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.30.2) on AKS 1.29.
```

* [GetOSOptions](https://github.com/Azure/AKS/issues/4261) will no longer be included in new AKS API versions starting with 2024-05-02. This API was used to get OS options that support Federal Information Process Standard (FIPS) in the specified subscription. If you're calling this API via the CLI, it will no longer be available in newer az aks extension versions. You can use an older version of the az aks extension, however this is not recommended. The CLI preview version supporting the 2024-05-02 preview API can be found [here](https://github.com/Azure/azure-cli-extensions/tree/main/src/aks-preview#released-version-and-adopted-api-version). Check the link for the release version.
For details on what AKS supported operating systems support Federal Information Process Standard (FIPS), see [aka.ms/aks/GetFIPSOSOptions](https://aka.ms/aks/GetFIPSOSOptions).

### Release Notes

> * Features:

Reviewer 4: need to add one more -
- Generally Available - [Azure Kubernetes Fleet Manager workload orchestration](https://learn.microsoft.com/azure/kubernetes-fleet/quickstart-resource-propagation)

  * Generally Available - AKS supports [disabling Windows OutboundNAT](https://learn.microsoft.com/azure/aks/nat-gateway#disable-outboundnat-for-windows).
  * Generally Available - [Automated Deployments](https://learn.microsoft.com/azure/aks/automated-deployments).
  * Generally Available - [Security patch channel](https://learn.microsoft.com/azure/aks/auto-upgrade-node-os-image?tabs=azure-cli) for VHD updates.

> * Preview Features  

Reviewer 4: need to add one more - 
- [Property-based scheduling](https://learn.microsoft.com/azure/kubernetes-fleet/intelligent-resource-placement) and [overrides](https://learn.microsoft.com/azure/kubernetes-fleet/cluster-resource-override) in Azure Kubernetes Fleet Manager.

  * Deployment Safeguards now supports [mutations in Enforcement mode](https://learn.microsoft.com/azure/aks/deployment-safeguards). 

>   * [Native sidecars](https://learn.microsoft.com/azure/aks/istio-native-sidecar) under Istio-based service mesh addon in AKS.

Reviewer 7: What feature is this? It just says native sidecars

  * [AKS Automatic](https://learn.microsoft.com/azure/aks/intro-aks-automatic). Visit the [AKS engineering blog](https://aka.ms/aks/hello-automatic) post.
  * [Node Initialization Taints](https://learn.microsoft.com/azure/aks/use-node-taints#use-node-initialization-taints-preview).
 
* Behavioral Changes:

>   * Allow [disabling NPM](https://learn.microsoft.com/azure/aks/use-network-policies#uninstall-azure-network-policy-manager-or-calico-preview) for existing clusters with "networkPolicy=none" for stable api version 2024-05-01. 

Reviewer 7: This is a preview feature looking at the link

  * [Node upgrade (reimage)](https://learn.microsoft.com/azure/aks/auto-upgrade-node-os-image) will wait for disk detach to complete (to prevent very slow disk detach).
  * Default [network policy](https://learn.microsoft.com/azure/aks/use-network-policies) is "networkPolicy=none" when network policy is not set on new clusters starting from API version 2024-05-01. This means that Network Policy Manager (NPM) is disabled. 

>   * [Advanced Container Networking Services](https://learn.microsoft.com/azure/aks/advanced-container-networking-services-overview) can be enabled on Cilium-enabled clusters with Kubernetes v1.29.0 or greater, and on Retina-enabled clusters with Kubernetes v1.21.0 or greater for Advanced Network Observability.

Reviewer 7: Preview feature

  * Customized apiserver subnet must be empty when migrating a cluster to enable [apiserver-vnet-integration](https://learn.microsoft.com/azure/aks/api-server-vnet-integration). If the subnet has resources in it, the migration won't be allowed.
   
* Bug Fixes:
  * Metrics server v0.6.3 will be used to prevent frequent OOMKills on Metric server.
  * [Allowing zonal NodeClaims to facilitate NodeClaims and node creation](https://github.com/Azure/karpenter-provider-azure/issues/339) on [Node Auto Provisioning](https://learn.microsoft.com/azure/aks/node-autoprovision) for workloads with zone affinity constraints. 
  * Fixed a bug where the [SSHAccess property of a node pool](https://learn.microsoft.com/azure/aks/manage-ssh-node-access) would be reset to LocalUser(SSHAccess:LocalUser) on a partial put. Henceforth, SSHAccess property will retain the current value (SSHAccess:current value).
  * Fixed bug where the eTag property in 2024-02-02 preview, 2024-03-02 preview, and 2024-04-02 preview APIs was returned with the wrong case (returned etag, should have been eTag).

>   * Patch version 1.28.9 fixes [Bug - OpenAPI handler fails on duplicated path](https://github.com/kubernetes/kubernetes/issues/122668). 

Reviewer 7: Is this release introducing patch version 1.28.9 ? If so that should be in the Features. Is this the only patch being added?

  * Updated cloud node manager to [v1.30.0](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.30.0) on AKS 1.30+, [v1.29.4](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.29.4) on AKS 1.29+, [v1.28.9](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.28.9) on AKS 1.28+, [v1.27.17](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.27.17) on AKS 1.27+. Refer [AKS version matrix for cloud node manager](https://github.com/kubernetes-sigs/cloud-provider-azure#aks-version-matrix).
  * Updated AKS App Routing operator image to [v0.2.3](https://github.com/Azure/aks-app-routing-operator/blob/main/CHANGELOG.md#023---2024-04-22).
  * Updated Azure File CSI driver to [v1.28.10](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.28.10)  on AKS 1.27, [v1.29.5](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.29.5)  on AKS 1.28, [v1.30.2](https://github.com/kubernetes-sigs/azurefile-csi-driver/releases/tag/v1.30.2) on AKS 1.29.
  * AKS Ubuntu 22.04 image has been updated to [AKSUbuntu-2204-202405.20.0](vhd-notes/aks-ubuntu/AKSUbuntu-2204/202405.20.0.txt).

>   * AKS Ubuntu 18.04 image has been updated to [AKSUbuntu-1804-202405.20.0](vhd-notes/aks-ubuntu/AKSUbuntu-1804/202405.20.0.txt).

Reviewer 7: 18.04 is deprecated and doesn't exist anymore

  * [AKS Automatic](https://learn.microsoft.com/azure/aks/intro-aks-automatic). Visit the [AKS engineering blog](https://aka.ms/aks/hello-automatic) post.
  * [Node Initialization Taints](https://learn.microsoft.com/azure/aks/use-node-taints#use-node-initialization-taints-preview).
  * [Advanced Container Networking Services](https://learn.microsoft.com/azure/aks/advanced-container-networking-services-overview) can be enabled on Cilium-enabled clusters with Kubernetes v1.29.0 or greater, and on Retina-enabled clusters with Kubernetes v1.21.0 or greater for Advanced Network Observability.
  * Allow [disabling NPM](https://learn.microsoft.com/azure/aks/use-network-policies#uninstall-azure-network-policy-manager-or-calico-preview) for existing clusters with "networkPolicy=none" for stable api version 2024-05-01. 
  * [Property-based scheduling](https://learn.microsoft.com/en-us/azure/kubernetes-fleet/intelligent-resource-placement) in Azure Kubernetes Fleet Manager.
  * [Cluster resource overrides](https://learn.microsoft.com/en-us/azure/kubernetes-fleet/cluster-resource-override) in Azure Kubernetes Fleet Manager.

>  

Reviewer 8: [Service Connector on AKS cluster is now public preview]([Tutorial: Connect to Azure OpenAI Service in AKS using a connection string (preview)](https://learn.microsoft.com/en-us/azure/service-connector/tutorial-python-aks-openai-connection-string?tabs=azure-portal)). It supports AKS pods connect to azure key vaut, database, Azure openAI, etc

Reviewer 8: @palma21 can we add service connector in aks release notes?

* Behavioral Changes:
  * [Node upgrade (reimage)](https://learn.microsoft.com/azure/aks/auto-upgrade-node-os-image) will wait for disk detach to complete (to prevent very slow disk detach).

>   * Default [network policy](https://learn.microsoft.com/azure/aks/use-network-policies) is "networkPolicy=none" when network policy is not set on new clusters starting from API version 2024-05-01. This means that Network Policy Manager (NPM) is disabled. 

Reviewer 9: But isn't it by default disabled?

Author: yes, it is we are calling it out explicitly, but I can remove it if it causes confusion

  * Customized apiserver subnet must be empty when migrating a cluster to enable [apiserver-vnet-integration](https://learn.microsoft.com/azure/aks/api-server-vnet-integration). If the subnet has resources in it, the migration won't be allowed.
   
* Bug Fixes:

>   * Metrics server v0.6.3 will be used to prevent frequent OOMKills on Metric server.

Reviewer 9: This one seems lacking context on reverting from v0.7, also isn't clear on for which minor versions.
